core/data/src/test/java/uk/gov/gchq/gaffer/data/element/EdgeTest.java

```diff
@@ -328,27 +350,151 @@ public void shouldSerialiseAndDeserialiseIdentifiers() throws SerialisationExcep
+                                            .source(new Vertex("1"))
+                                            .destination(new Vertex("2"))
+                                            .build();
+
+        // Then
+        assertThat(edge.getSource(), equalTo(new Vertex("1")));
+        assertThat(edge.getDestination(), equalTo(new Vertex("2")));
+    }
+
+    @Test
+    public void shouldSwapVerticesIfSourceIsGreaterThanDestination_comparable() {
+        // Given
+        final Edge edge = new Edge.Builder().group(TestGroups.EDGE)
+                                            .directed(false)
+                                            .source(new Integer(2))
+                                            .destination(new Integer(1))
+                                            .build();
+
+        // Then
+        assertThat(edge.getSource(), equalTo(new Integer(1)));
+        assertThat(edge.getDestination(), equalTo(new Integer(2)));
+    }
+
+    @Test
+    public void shouldNotSwapVerticesIfSourceIsLessThanDestination_comparable() {
+        // Given
+        final Edge edge = new Edge.Builder().group(TestGroups.EDGE)
+                                            .directed(false)
+                                            .source(new Integer(1))
+                                            .destination(new Integer(2))
+                                            .build();
+
+        // Then
+        assertThat(edge.getSource(), equalTo(new Integer(1)));
+        assertThat(edge.getDestination(), equalTo(new Integer(2)));
+    }
+
+    @Test
+    public void shouldFailToConsistentlySwapVerticesWithNoToStringImplementation() {
+        // Given
+        final List<Edge> edges = new ArrayList<>();
+        final List<Vertex2> sources = new ArrayList<>();
+        final List<Vertex2> destinations = new ArrayList<>();
+
+        // Create a load of edges with Vertex2 objects as source and destination.
+        // Vertex2 has no toString method and does not implement Comparable, so
+        // this should result in Edges being created with different sources and
+        // destinations.
+        for (int i = 0;  i < 1000; i ++) {
+            final Vertex2 source = new Vertex2("1");
+            final Vertex2 destination = new Vertex2("2");
+
+            sources.add(source);
+            destinations.add(destination);
+        }
+
+        for (int i = 0;  i < 1000; i ++) {
+            final Edge edge = new Edge.Builder().group(TestGroups.EDGE)
+                                                .directed(false)
+                                                .source(sources.get(i))
+                                                .destination(destinations.get(i))
+                                                .build();
+
+            edges.add(edge);
+        }
+
+        // Then
+        assertThat(edges.stream().distinct().count(), greaterThan(1L));
+        assertThat(edges.stream().distinct().count(), greaterThan(1L));
```

Comment:
These 2 lines are the same. Was the intention that the first line counted the number of unique values of the source and the second line counted the number of destinations?